chainio/txmgr/txmgr.go

```diff
@@ -0,0 +1,233 @@
+package txmgr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	FallbackGasTipCap = big.NewInt(15000000000)
+)
+
+// We are taking inspiration from the optimism TxManager interface
+// https://github.com/ethereum-optimism/optimism/blob/develop/op-service/txmgr/txmgr.go
+
+type TxManager interface {
+	// Send is used to send a transaction
+	// It takes an unsigned transaction and then signs it before sending
+	// It also takes care of nonce management and gas estimation
+	Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error)
+
+	// Call is used to call a contract method
+	Call(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+
+	// GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+	// this to generate the transaction without actually sending it
```

Comment:
The typical sequence would be 
1. call GetNoSendTxOpts()
2. pass this to some abigen contract to generate the tx
3. pass the tx to Send() above to sign it and send it
right?

Feels like we should add this as a comment somewhere above the interface probably?

Reply:
We discussed this offline. Let's add this sequence as a comment to the txmgr, and also add a uml diagram illustrating the flow in the README

Reply:
added a diagram 9aa78223e2df8a796d342c8ad71489270af2e813